src/lib/alloc.c

```diff
@@ -579,10 +590,12 @@ int mm_pm_context_save(struct dma_copy *dc, struct dma_sg_config *sg)
 		return ret;
 
 	/* copy system memory contents to SG */
-	ret = dma_copy_to_host(dc, sg, offset + ret,
-		(void *)memmap.system.heap, (int32_t)(memmap.system.size));
-	if (ret < 0)
-		return ret;
+
+	/* TODO: does it work? heap ptr points to free location, not to base */
+//	ret = dma_copy_to_host(dc, sg, offset + ret,
```

Comment:
ok, the context save/restore can be removed since it's not used atm. This should simplify the code.